Sources/InteroperabilityTests/InteroperabilityTestCases.swift

```diff
@@ -0,0 +1,721 @@
+    let stream = AsyncStream(Int.self) { streamContinuation in
+      continuation = streamContinuation
+    }
+    #endif
+    let request = ClientRequest.Stream { writer in
+      for try await id in stream {
+        var message = Grpc_Testing_StreamingOutputCallRequest()
+        switch id {
+        case 1:
+          message.responseParameters = [
+            Grpc_Testing_ResponseParameters.with {
+              $0.size = 31_415
+            }
+          ]
+          message.payload = Grpc_Testing_Payload.with {
+            $0.body = Data(count: 27_182)
+          }
+        case 2:
+          message.responseParameters = [
+            Grpc_Testing_ResponseParameters.with {
+              $0.size = 9
+            }
+          ]
+          message.payload = Grpc_Testing_Payload.with {
+            $0.body = Data(count: 8)
+          }
+        case 3:
+          message.responseParameters = [
+            Grpc_Testing_ResponseParameters.with {
+              $0.size = 2_653
+            }
+          ]
+          message.payload = Grpc_Testing_Payload.with {
+            $0.body = Data(count: 1_828)
+          }
+        case 4:
+          message.responseParameters = [
+            Grpc_Testing_ResponseParameters.with {
+              $0.size = 58_979
+            }
+          ]
+          message.payload = Grpc_Testing_Payload.with {
+            $0.body = Data(count: 45_904)
+          }
+        default:
+          return
+        }
+        try await writer.write(message)
+      }
+    }
+    continuation.yield(1)
+    try await testServiceClient.fullDuplexCall(request: request) { response in
+      var id = 1
+      for try await message in response.messages {
+        switch id {
+        case 1:
+          try assertEqual(message.payload.body, Data(count: 31_415))
+        case 2:
+          try assertEqual(message.payload.body, Data(count: 9))
+        case 3:
+          try assertEqual(message.payload.body, Data(count: 2_653))
+        case 4:
+          try assertEqual(message.payload.body, Data(count: 58_979))
+        default:
+          try assertFailure("We should only receive messages with ids between 1 and 4.")
+        }
+
+        // Add the next id to the continuation.
+        id += 1
+        continuation.yield(id)
+      }
+    }
+  }
+}
+
+/// This test verifies that streams support having zero-messages in both directions.
+///
+/// Server features:
+/// - FullDuplexCall
+///
+/// Procedure:
+/// 1. Client calls FullDuplexCall and then half-closes
+///
+/// Client asserts:
+/// - call was successful
+/// - exactly zero responses
+@available(macOS 13.0, iOS 16.0, watchOS 9.0, tvOS 16.0, *)
+struct EmptyStream: InteroperabilityTest {
+  func run(client: GRPCClient) async throws {
+    let testServiceClient = Grpc_Testing_TestService.Client(client: client)
+
+    let request = ClientRequest.Stream { writer in
+      let message = Grpc_Testing_StreamingOutputCallRequest()
+      try await writer.write(message)
```

Comment:
The test wants an _empty_ stream, this is a stream of _one_ message. In this case we just want the request to be `ClientRequest.Stream { _ in }`